routing/router.go

```diff
@@ -1741,16 +1741,86 @@ type routingMsg struct {
 	err chan error
 }
 
+// RouteRequest contains the parameters for a pathfinding request.
+type RouteRequest struct {
+	// Source is the node that the path originates from.
+	Source route.Vertex
+
+	// Target is the node that the path terminates at.
+	Target *route.Vertex
```

Comment:
This is in preparation for a future commit where we want to allow a nil `Target` value in the case where the end user has provided a blinded path (where we can get the target from the final hop). 

This is a somewhat stylistic choice, because we could also ask the user to add a target *and* a blinded route (and then check that they're equal) _but_ that seemed like more code + more user hassle so I went with mutually exclusive instead.